src/qt/modaloverlay.cpp

```diff
@@ -172,3 +191,21 @@ void ModalOverlay::closeClicked()
     showHide(true);
     userClosed = true;
 }
+
+void ModalOverlay::pauseClicked()
+{
+    Q_EMIT requestVerificationPauseOrResume();
+}
+
+void ModalOverlay::setPauseResumeState(bool pauseActive)
+{
+    verificationPauseActive = pauseActive;
+    updatePauseState(pauseActive);
+}
+
+void ModalOverlay::updatePauseState(bool pauseActive)
+{
+    ui->labelNumberBlocksRequested->setText((pauseActive ? "Finish downloading blocks" : "Blocks requested from peers"));
```

Comment:
I'd suggest changing this string to just "Blocks in download queue" or similar. Then we don't have to worry about switching strings between paused and downloading.